sklearn/feature_extraction/text.py

```diff
@@ -1103,6 +1103,16 @@ def __init__(
         self.min_df = min_df
         if max_df < 0 or min_df < 0:
             raise ValueError("negative value for max_df or min_df")
+        if not isinstance(min_df, numbers.Integral) and min_df > 1.0:
```

Comment:
I think we could do,
```suggestion
        if (not isinstance(min_df, numbers.Integral) and 
            (min_df < 1.0  or min_df > 1.0)):
```
and remove the above check since the error message there is not very helpful. Same for `max_df`. This might need adjusting tests though.

Reply:
Hi @rth, thanks for your feedback! Do you mean to check min_df < **0**?

I'd be glad to make those changes but it's my understanding that we should still be checking for negative integer values and not just float.

We can remove the above check like you're suggesting but rewrite the new check this way instead:
```suggestion
        if min_df < 0 or
            (not isinstance(min_df, numbers.Integral) and min_df > 1.0):
```

Same for max_df.

Thanks!

Reply:
Yes, you are right the parenthesis was in the wrong place. Though yes, for an int the current message would not be ideal. OK, let's leave unchanged as it is.